src/cache_invalidate.c

```diff
@@ -1,36 +1,43 @@
-/*
- *	Notes on the way caching works: Since out caches are stored in per-process
- *	(per-backend memory), we have to have a way to propagate invalidation
- *	messages to all backends, for that we use the Postgres relcache
- *	mechanism. Relcache is a cache that keeps internal info about
- *	relations(tables).  Postgres has a mechanism for registering for relcache
- *	invalidation events that are propagated to all backends:
- *	CacheRegisterRelcacheCallback(). We register inval_cache_callback() with
- *	this mechanism and route all invalidation messages through it to the correct
- *	cache invalidation functions.
- *
- *	The plan for our caches is to use (abuse) this mechanism to serve as a
- *	notify to invalidate our caches.  Thus, we create proxy tables for each
- *	cache we use and attach the invalidate_relcache_trigger trigger to all
- *	tables whose changes should invalidate the cache. This trigger will
- *	invalidate the relcache for the proxy table specified as the first argument
- *	to the trigger (see cache.sql).
- */
-
 #include <postgres.h>
 #include <access/xact.h>
 #include <utils/lsyscache.h>
+#include <utils/inval.h>
 #include <catalog/namespace.h>
-#include <miscadmin.h>
 #include <commands/trigger.h>
-#include <commands/event_trigger.h>
 #include <nodes/nodes.h>
-#include <utils/inval.h>
-#include <unistd.h>
+#include <miscadmin.h>
 
 #include "hypertable_cache.h"
 #include "catalog.h"
 #include "extension.h"
+#include "cache_invalidate.h"
+
+/*
+ * Notes on the way cache invalidation works.
+ *
+ * Since our caches are stored in per-process (per-backend memory), we need a
+ * way to signal all backends that they should invalidate their caches. For this
+ * we use the PostgreSQL relcache mechanism that propagates relation cache
+ * invalidation events to all backends. We register a callback with this
+ * mechanism to recieve events on all backends whenever a relation cache entry
+ * is invalidated.
+ *
+ * To know which events should trigger invalidation of our caches, we use dummy
+ * (empty) tables. We can trigger relcache invalidation events for these tables
+ * to signal other backends. If the received table OID is a dummy table, we know
+ * that this is an event that we care about.
+ *
+ * Caches for catalog tables should be invalidated on:
+ *
+ * 1. INSERT/UPDATE/DELETE on a catalog table
+ * 2. Aborted transactions that taint the caches
+ *
+ * Generally, INSERTS do not warrant cache invalidation, unless it is an insert
```

Comment:
I am not sure what is premature about it, unless everything which is not brute force is premature? Inserting a new hypertable should have no effect on existing hypertables.

Reply:
Actually for hypertables this is strictly not true because there are negative cache entries (cache entries that say that a table is /not/ a hypertable).

Reply:
This is handled by the INSERT on `dimension` when you do `create_hypertable()`.

Reply:
Btw, made `HYPERTABLE` and `DIMENSION` handling the same, w.r.t. cache invalidation